service/debug/debug_service.go

```diff
@@ -48,28 +50,38 @@ func (s *DebugService) Start() error {
 	s.Publish("memstats", Func(memstats))
 
 	go func() {
-		// Prefer to listen on addr, but will try to bind to the next 9 ports
-		// in case you have multiple services running on the same host.
-		for i := 0; i < 10; i++ {
-			host, portStr, _ := net.SplitHostPort(addr)
-			port, _ := strconv.Atoi(portStr)
-			port += i
-			addr := net.JoinHostPort(host, fmt.Sprint(port))
-
+		defaultAddr := s.Config.GetDebugServiceAddr()
```

Comment:
is this the default address? It looks like if this is set then it is the specified address and we won't use the default?